x-pack/test/api_integration/apis/siem/feature_controls.ts

```diff
@@ -0,0 +1,297 @@
+        expectGraphQL404(graphQLResult);
+
+        const graphQLIResult = await executeGraphIQLRequest(username, password);
+        expectGraphIQL404(graphQLIResult);
+      } finally {
+        await security.role.delete(roleName);
+        await security.user.delete(username);
+      }
+    });
+
+    it('APIs can be accessed user with global "all" and logstash-* "read" privileges', async () => {
+      const username = 'global_all';
+      const roleName = 'global_all';
+      const password = `${username}-password`;
+      try {
+        await security.role.create(roleName, {
+          elasticsearch: {
+            indices: [
+              {
+                names: ['logstash-*'],
+                privileges: ['read', 'view_index_metadata'],
+              },
+            ],
+          },
+          kibana: [
+            {
+              base: ['all'],
+              spaces: ['*'],
+            },
+          ],
+        });
+
+        await security.user.create(username, {
+          password,
+          roles: [roleName],
+          full_name: 'a kibana user',
+        });
+
+        const graphQLResult = await executeGraphQLQuery(username, password);
+        expectGraphQLResponse(graphQLResult);
+
+        const graphQLIResult = await executeGraphIQLRequest(username, password);
+        expectGraphIQLResponse(graphQLIResult);
+      } finally {
+        await security.role.delete(roleName);
+        await security.user.delete(username);
+      }
+    });
+
+    // this could be any role which doesn't have access to the infra feature
+    it(`APIs can't be accessed by user with dashboard "all" and logstash-* "read" privileges`, async () => {
+      const username = 'dashboard_all';
+      const roleName = 'dashboard_all';
+      const password = `${username}-password`;
+      try {
+        await security.role.create(roleName, {
+          elasticsearch: {
+            indices: [
+              {
+                names: ['logstash-*'],
+                privileges: ['read', 'view_index_metadata'],
+              },
+            ],
+          },
+          kibana: [
+            {
+              feature: {
+                dashboard: ['all'],
+              },
+              spaces: ['*'],
+            },
+          ],
+        });
+
+        await security.user.create(username, {
+          password,
+          roles: [roleName],
+          full_name: 'a kibana user',
+        });
+
+        const graphQLResult = await executeGraphQLQuery(username, password);
+        expectGraphQL404(graphQLResult);
+
+        const graphQLIResult = await executeGraphIQLRequest(username, password);
+        expectGraphIQL404(graphQLIResult);
+      } finally {
+        await security.role.delete(roleName);
+        await security.user.delete(username);
+      }
+    });
+
+    describe('spaces', () => {
+      // the following tests create a user_1 which has siem read access to space_1, logs read access to space_2 and dashboard all access to space_3
```

Comment:
Infra has two features `infrastructure` and `logs`, but you all only have one.  I'd recommend changing these tests to ensure that user_1 has read access to siem in space_1, and dashboard all access in space_2. You shouldn't need space_3.